src/runtime/utils/url.ts

```diff
@@ -5,7 +5,17 @@ import type { ModuleOptionsNormalized } from '../types'
 import { useRequestEvent, useNuxtApp, abortNavigation, useAuthState } from '#imports'
 
 export const getRequestURL = (includePath = true) => getURL(useRequestEvent()?.node.req, includePath)
-export const joinPathToApiURL = (path: string) => joinURL(useAuthState()._internal.baseURL, path)
+export function joinPathToApiURL (path: string) {
+  const authStateInternal = useAuthState()._internal
+
+  // For internal calls, use a different base
+  // https://github.com/sidebase/nuxt-auth/issues/742
+  const base = path.startsWith('/')
+    ? authStateInternal.pathname
+    : authStateInternal.baseURL
```

Comment:
One thing I worry about: Relying on `/` to determine whether a path is internal -> is there another way? I think there may be / should be, as we know which endpoints we are registering

Reply:
This is exactly what $fetch does under the hood — it looks at the first character in the passed string and if it is `/`, does an internal dispatch.

I don't know of a way to check internal API routes otherwise, so suggestions would be welcome

Reply:
Ah I see - in that case, let's do it like this then (: